### PR TITLE
LPS-186817 Method getItems(httpServletRequest, filter, pagination, sort) had the order of the parameters changed

### PR DESCRIPTION
**Pull request by michaelmdelcavalcanti**
Hi Kevin,
The purpose of this check is to sort the variables of the getItems and getItemsCount methods and also their method calls.
I had some problems during the implementation that made me take some questionable directions.

1 - I created a regex that checks if the interface was implemented or not, in the client code I saw situations where the interface was imported but not implemented, it was used for a different purpose.
2 - The client uses the interface in a different way than the conventional one, as can be seen in the test file.
3 - The getVariableTypeName method does not work correctly with variables with type that have generics(so i tried a workaround solution), as can be seen in the test file.

I am waiting for your revision and suggestions.

### CODE REVIEW

File: modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/JavaUpgradeFDSDataProviderCheck.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+					methodCallGetItemsCountMatcher.group(2),
+					newOrderVariablesMethodCall);
+
+				content = StringUtil.replace(
+					content, methodCallGetItemsCountMatcher.group(0), newLine);
+			}
+		}
+
+		return content;
+	}
```
Comment (kevhlee) [inline modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/JavaUpgradeFDSDataProviderCheck.java:134]: For method calls, you fetch the parameters using `JavaSourceUtil.getParameterList()`.

**Review by kevhlee — CHANGES_REQUESTED** [review]
To summarize my comments, the check needs to leverage more of what is already available in SF. In the future, when writing checks, if there is something you do not know how to do efficiently, look at other checks for reference. Most of the time, you'll find that there is some utility method that will help you achieve what you need, at least partially.

Also, regarding `getVariableTypeName()` not working, there is another `getVariableTypeName()` that takes a boolean flag for capturing array/collection types. Have you tried using that?

File: modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/expected/upgrade/UpgradeJavaFDSDataProviderCheck.testjava [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	@Override
+	public List getItems(FDSKeywords fdsKeywords, FDSPagination pagination, HttpServletRequest httpServletRequest, Sort sort)
+		throws PortalException {
+
+		return _original.getItems(fdsKeywords, pagination, httpServletRequest, sort
+		).stream(
+		).map(
+		).collect(
+			Collectors.toList()
+		);
```
Comment (kevhlee) [inline modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/expected/upgrade/UpgradeJavaFDSDataProviderCheck.testjava:33]: This is a strange usage/example. Technically, it is not valid Java code. Also, line 25 breaks the max 80 column rule. For this check, the 80 column rule is not irrelevant to the check, so we need to format it.

**Review by kevhlee — CHANGES_REQUESTED** [review]

**Comment by michaelmdelcavalcanti** [comment]
ci:test:relevant

**Comment by michaelmdelcavalcanti** [comment]
ci:test:sf

**Comment by michaelmdelcavalcanti** [comment]
ci:test:relevant

**Comment by michaelmdelcavalcanti** [comment]
ci:test:sf

**Comment by michaelmdelcavalcanti** [comment]
ci:test:relevant

File: modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaFDSDataProviderCheck.java [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+
+			Matcher methodCallGetItemsCountMatcher =
+				_methodCallGetItemsCountPattern.matcher(javaMethodContent);
+
+			while (methodCallGetItemsCountMatcher.find()) {
+				String methodCall = JavaSourceUtil.getMethodCall(
+					javaMethodContent, methodCallGetItemsCountMatcher.start());
+
+				List<String> parameterList = JavaSourceUtil.getParameterList(
+					methodCall);
```
Comment (kevhlee) [inline modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaFDSDataProviderCheck.java:107]: What happens if the list is empty?

**Review by kevhlee — CHANGES_REQUESTED** [review]

File: modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/upgrade/UpgradeJavaFDSDataProviderCheck.testjava [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+/**
+ * @author Michael Cavalcanti
+ */
+public class UpgradeJavaFDSDataProviderCheck implements FDSDataProvider {
+
+	@Override
+	public List getItems(
+		HttpServletRequest httpServletRequest, FDSKeywords fdsKeywords,
+		FDSPagination pagination, Sort sort)
+		throws PortalException {
```
[the remark this answers is not in the document]
Comment (michaelmdelcavalcanti) [reply modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/upgrade/UpgradeJavaFDSDataProviderCheck.testjava:28]: I had run SF, but realized that it was not formatted correctly, and investigated and identified that my IDE was causing some formatting problems.

File: modules/util/source-formatter/src/main/resources/sourcechecks.xml [hunk 5]
```diff
@@ -1592,11 +1592,17 @@
 		<check name="UpgradeJavaCheck">
 			<category name="Upgrade" />
 			<description name="Performs upgrade checks for `java` files" />
+			<weight>2</weight>
```
[the remark this answers is not in the document]
Comment (michaelmdelcavalcanti) [reply modules/util/source-formatter/src/main/resources/sourcechecks.xml:1595]: The FDSDataProvider class is replacing the ClayDataSetDataProvider class, this replacement is being implemented by UpgradeJavaCheck, as the check was created based on FDSDataProvider, it is necessary that UpgradeJavaCheck runs first.

File: modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaFDSDataProviderCheck.java [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+		String content, String fileContent, String methodCall) {
+
+		String variableName = methodCall.substring(
+			0, methodCall.indexOf(StringPool.PERIOD));
+
+		return getVariableTypeName(
+			content, fileContent, variableName, true
+		).contains(
+			"FDSDataProvider"
+		);
```
[the remark this answers is not in the document]
Comment (michaelmdelcavalcanti) [reply modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaFDSDataProviderCheck.java:217]: Ok, I'll fix it.

File: modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaFDSDataProviderCheck.java [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+				"com.liferay.frontend.data.set.provider.FDSDataProvider")) {
+
+			return content;
+		}
+
+		content = StringUtil.replace(
+			content, javaClass.getContent(), _checkMethods(javaClass));
+
+		content = StringUtil.replace(
+			content, javaClass.getContent(), _checkMethodCalls(javaClass));
```
[the remark this answers is not in the document]
Comment (michaelmdelcavalcanti) [reply modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaFDSDataProviderCheck.java:59]: Yes, I'll fix it.

**Comment by michaelmdelcavalcanti** [comment]
ci:test:sf

**Comment by michaelmdelcavalcanti** [comment]
ci:test:relevant

**Comment by michaelmdelcavalcanti** [comment]
ci:stop

**Comment by michaelmdelcavalcanti** [comment]
ci:test:sf

**Comment by michaelmdelcavalcanti** [comment]
ci:test:relevant

File: modules/util/source-formatter/src/main/resources/sourcechecks.xml (hunk 5 above)
[the remark this answers is not in the document]
Comment (kevhlee) [reply modules/util/source-formatter/src/main/resources/sourcechecks.xml:1595]: Ok, got it.

**Comment by kevhlee** [comment]
Taking a look.

**Comment by kevhlee** [comment]
See here: <https://github.com/kevhlee/liferay-portal/pull/97>

**Pull request closed without merging**